### DISCUSSION TITLE
InputRules question

### DISCUSSION THREAD

**Discussion by apg-dev** [ueberdosis/tiptap#688] [General Questions]
Hello community. Apologies for this basic question.

I recently discovered TipTap while trying to build an editor that allows me to add customizations such as auto-correcting my various typos (eg. auto-converting “figur eout” to “figure out” and plenty of other multi-word typos that standard Mac OS keyboard shortcuts won’t allow). In addition to building a custom “auto-correct” dictionary, I also want to use certain keyboard shortcuts such as “/t” to automatically insert the time. And eventually I’d like to implement the ability to link notes to one another via double-bracket syntax (e.g. text within Note One that links to [[Note Two]]).

I’m sure all of this is do-able with inputrules, but I can’t figure out how to do even the basic typo correction stuff. I’ve been reviewing TipTap’s documentation, which is great - just wishing there was more. And looking at ProseMirror’s docs which are comprehensive but very hard to follow for a person with my limited Vue and JS skills (I’m coming from the C# MVC world).

Appreciate any help or guidance the community can offer. Thank you!

**Comment by phxtho** [comment]
Was also looking into creating a roam-like note tool in Vue similarly stuck on creating extensions.

**Comment by clarkmcc** [comment]
I assume this library has changed a lot in the last few years. Is there a standard way to do this Roam-style double bracket syntax?
